### PR TITLE
`azurerm_databricks_workspace` - Fix propagation of tags to connected resources

### LINKED ISSUE #11395: azurerm_databricks_workspace tags argument always forces replacement of databricks_workspace resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.15.0 
AzureRM version = ">= 2.26"

### Affected Resource(s)

* `azurerm_databricks_workspace`

### Terraform Configuration Files

```hcl
#
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = ">= 2.26"
    }
  }

  required_version = ">= 0.14.9"
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "rg" {
  name     = "testrg"
  location = "eastus2"
}
resource "azurerm_databricks_workspace" "workspace" {
  name                = "testws"
  resource_group_name = "testrg"
  location            = "eastus2"
  sku                 = "standard"

  tags = {
    Environment = "terraformtest"
	WorkspacePatch = "terraformtest"
  }
}
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Expected Behaviour
If user changes the tags param and terraform apply - it should only update the tags i.e it should call PATCH

### Actual Behaviour
Tags change forces to destroy and re-create the complete workspace object. It is calling PUT even though only tags are changed

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by WodansSon**
fixes #11395

### CODE REVIEW

**Review by JeetSharan — APPROVED** [review]

File: azurerm/internal/services/databricks/databricks_workspace_resource.go [hunk 1]
```diff
@@ -184,6 +184,8 @@ func resourceDatabricksWorkspaceCreateUpdate(d *schema.ResourceData, meta interf
 	customParamsRaw := d.Get("custom_parameters").([]interface{})
 	customParams := expandWorkspaceCustomParameters(customParamsRaw)
 
+	// Including the Tags in the workspace parameters will update the tags on
+	// the workspace only
```
Comment (katbyte) [inline azurerm/internal/services/databricks/databricks_workspace_resource.go:188]: this seems like an API bug? could we open a issue on the sdk/specs and link to it here?

**Review by katbyte — APPROVED** [review]
Aside from one comment this LGTM 👍

File: azurerm/internal/services/databricks/databricks_workspace_resource.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/internal/services/databricks/databricks_workspace_resource.go:188]: it kind of is, although tbh this is another case where it'd be beneficial to split the Create/Update here longer-term?

Comment (WodansSon) [reply azurerm/internal/services/databricks/databricks_workspace_resource.go:188]: This totally is an API bug and I have opened issue [14571](https://github.com/Azure/azure-sdk-for-go/issues/14571) to track this bug.

**Comment by WodansSon** [comment]
The two test cases that failed are known issues:
image

**Pull request merged**